modules/terraform/apply_test.go

```diff
@@ -64,5 +58,40 @@ func TestApplyWithErrorWithRetry(t *testing.T) {
 
 	out := InitAndApply(t, options)
 
-	assert.Contains(t, out, "This is the first run, exiting with an error")
+	require.Contains(t, out, "This is the first run, exiting with an error")
+}
+func TestTgApplyAllTgError(t *testing.T) {
+	t.Parallel()
+
+	testFolder, err := files.CopyTerragruntFolderToTemp("../../test/fixtures/terragrunt/terragrunt-no-error", t.Name())
```

Comment:
Terragrunt assumes a `terraform.tfvars` file containing a terragrunt block with a source ref. I think you need this at  a minimum for the no error example.